src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,157 @@
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidPasswordLength_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("qwert");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAgeNull_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidAge_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAllInvalidFields_NotOk() {
+        user.setLogin("Steve");
+        user.setPassword("qwert");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidAgeHigherThanIntegerMaxValue_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(Integer.MAX_VALUE + 10);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAllValidFields_Ok() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(20);
+        User returnedUser = registrationService.register(user);
+        assertEquals(user, returnedUser);
+    }
+
+    @Test
+    void getUser_ByLogin_Ok() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(20);
+        registrationService.register(user);
+        User getUser = storageDao.get(user.getLogin());
+        assertEquals(user, getUser);
+    }
+
+    @Test
+    void getUser_ByNotExistLogin_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(20);
+        registrationService.register(user);
+        User getUser = storageDao.get("Steve1234");
+        assertNotEquals(user, getUser);
+    }
+
+    @Test
+    void registerUser_WithExistLogin_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(20);
+        registrationService.register(user);
+        User usertoAdd = new User();
+        usertoAdd.setLogin("Steve123");
+        usertoAdd.setPassword("valid123");
+        usertoAdd.setAge(25);
+        User getUser = storageDao.get(user.getLogin());
+        assertNotEquals(usertoAdd, getUser);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
```

Comment:
The test `registerUser_WithExistLogin_NotOk` is not correct. The `assertNotEquals` check on line 152 is unnecessary because you haven't added `usertoAdd` to the storage yet. Additionally, the `assertThrows` on line 153 should be testing the registration of `usertoAdd`, not `user`, since `user` has already been added to the storage and the purpose is to test duplicate registration.